### ISSUE TITLE
Swagger method names nicePascalName is not a nice name.

### ISSUE THREAD

**Issue by xenocons** [fsprojects/SwaggerProvider#45]
Unsure if bug or feature, but the generated method names (https://github.com/fsprojects/SwaggerProvider/blob/2184d9e625056c1ce2070f3beca36c6df91390ae/src/SwaggerProvider.DesignTime/OperationCompiler.fs#L208) are fed through `nicePascalName`. This causes a separation between what is in the JSON schema and the function name that is generated.

E.g. `"operationId":"Application_GetLatestApplicationDetails"` in the JSON schema gets turned into `nicePascalName "Application_GetLatestApplicationDetails"` which subsequently returns `ApplicationGetLatestApplicationDetails` which is less ideal. 

`nicePascalName "Application_GetLatestApplicationDetails" = nicePascalName "ApplicationGet_LatestApplicationDetails"` returns true, even though they are separate controllers and separate functions.

I would expect that the `operationId` in the JSON directly maps to a function in the generated results of the TP.

**Comment by sergey-tihon** [comment]
The initial idea was to generate names that will play nice for `C#` users, because it is generated type provider and we have no stoppers to be C# friendly,

That's why I try to avoid `_` in function names, `nicePascalName` is reused from `FSharp.Data`.

How to you generate swagger schema in you case? Do you use `Swashbuckle` or you schema is hard-written?

During the testing agains [APIs.guru](https://apis.guru/) schemas, i found that I could not rely on `operationId` for all cases

**Comment by xenocons** [comment]
Hmm , Interesting, I generate my schema using swashbuckle (see https://github.com/fsprojects/SwaggerProvider/issues/46#issuecomment-240463272), I would *like* Swagger to generate static classes (Modules) for controllers, this would be similar to the C# approach (see below). 

In Visual Studio 2015 if we add a C# WebAPI project (C# WebApp -> WebAPI), then right click the new WebAPI project -> Add -> REST Api Client, and enter in the JSON from our API (say http://localhost:48213/swagger/docs/v1), we actually see that the C# codegen takes the controller name and uses this as a partial class.

**Comment by sergey-tihon** [comment]
This feature is implemented and released in [0.10.0-alpha7](https://www.nuget.org/packages/SwaggerProvider/0.10.0-alpha7) but turned off by default.

In order to turn it on, you need to add parameter `IgnoreControllerPrefix = false`
For each controller will be generated separate `*Client` class 

### Sample
```fsharp
type GI = SwaggerProvider<"https://api.apis.guru/v2/specs/gettyimages.com/3/swagger.json", IgnoreControllerPrefix = false>

let c1 = GI.ArtistsClient()
let c2 = GI.UsageClient()
```

<img width="1141" alt="2017-12-09_2203" src="https://user-images.githubusercontent.com/1197905/33808447-882b0fa6-ddf7-11e7-83a9-4da7c2f17252.png">
